src/mito2/src/memtable/time_series.rs

```diff
@@ -369,13 +362,46 @@ struct Iter {
     series: Arc<SeriesRwLockMap>,
     projection: HashSet<ColumnId>,
     last_key: Option<Vec<u8>>,
-    predicate: Vec<Arc<dyn PhysicalExpr>>,
+    predicate: Vec<SimpleFilterEvaluator>,
     pk_schema: arrow::datatypes::SchemaRef,
-    primary_key_builders: Vec<Box<dyn MutableVector>>,
+    // primary_key_builders: Vec<Box<dyn MutableVector>>,
+    pk_datatypes: Vec<ConcreteDataType>,
     codec: Arc<McmpRowCodec>,
     metrics: Metrics,
 }
 
+impl Iter {
+    pub(crate) fn new(
+        metadata: RegionMetadataRef,
+        series: Arc<SeriesRwLockMap>,
+        projection: HashSet<ColumnId>,
+        predicate: Option<Predicate>,
+        pk_schema: arrow::datatypes::SchemaRef,
+        pk_datatypes: Vec<ConcreteDataType>,
+        codec: Arc<McmpRowCodec>,
+    ) -> Self {
+        let simple_filters = predicate
+            .map(|p| {
+                p.exprs()
+                    .iter()
+                    .filter_map(|f| SimpleFilterEvaluator::try_new(f.df_expr()))
+                    .collect::<Vec<_>>()
+            })
+            .unwrap_or_default();
```

Comment:
Do we have to fallback to physical expr evaluator when given predicates cannot be converted to simple evaluator?

Reply:
Predicates here are not required to be accurate. We can ignore them if evaluating on physical expr is expansive. And the filter behavior of parquet and memtable are aligned now.